### PR TITLE
refactor(binding): reduced code size

### PR DESCRIPTION
**Pull request by jods4**
Members from exported classes and strings do not minify :(

I rewrote the creating of the mapping to be more compact, easily shaving ~200 bytes.

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
@jdanyow Can you review this?

File: src/attribute-map.js [hunk 1]
```diff
@@ -9,43 +9,39 @@ export class AttributeMap {
...
+    .forEach(n => {
+      this.register('td', n, n);
+      this.register('th', n, n);
+    });    
   }
 
   /**
    * Maps a specific HTML element attribute to a javascript property.
    */
   register(elementName, attributeName, propertyName) {
```
Comment (jdanyow) [inline src/attribute-map.js:35]: I think the test suite is missing tests that use the equivalent of `element.nodeName`/`element.tagName` which is upper-case. Could you add a test for `expect(attributeMap.map('LABEL', 'for')).toBe('htmlFor');`?

**Review by jdanyow — CHANGES_REQUESTED** [review]
Totally understand the motivation for this change (aurelia/framework#692). It comes at the expense of the code-base's readability/maintainability. This might be ok, I just want to make sure we discuss what the guidelines should be for these types of changes.

Is there a certain "byte reduction" threshold that must be met before accepting a PR that obfuscates the code but reduces the size?

Are there more "big impact" things we could do before getting into micro optimizations like this? For example, we're planning on converting the code-base to TypeScript to take advantage of smaller transpiled output due to the lack of classCallCheck and more terse class method output (compare [babel's](https://github.com/aurelia/templating-binding/blob/58fb98f6f1699ac839cf5d1d439694786b961eeb/dist/amd/aurelia-templating-binding.js#L100) with [typescript's](http://www.typescriptlang.org/play/#src=export%20class%20AttributeMap%20%7B%0D%0A%20%20%20%20registerUniversal(attributeName%2C%20propertyName)%20%7B%0D%0A%0D%0A%20%20%20%20%7D%0D%0A%7D)).

My vote would be to continue working on the big impact items before doing things like this but I don't want to stand in the way of progress if everyone else agrees this is the course we should take.

Thoughts?

cc @EisenbergEffect

**Comment by jods4** [comment]
@jdanyow 
I agree that maintainability is more important than code size. 
The two are not incompatible and often smaller code can be simpler code (e.g. we have Promise spaghetti 🍝 going on in a few places). I am really trying to _not_ obfuscate anything for the sake of size.

In this particular case, I think the result is simple and clear enough, but that's subjective. 
The only thing that I find not intuitive is that attributes are lowercased by `register`, which is why I left a comment.

If you think the code is not clear enough then we can drop this PR (or maybe just keep the `register` improvement), it's only a 200 bytes reduction after all. 

When I saw it I thought it was 200 _easy_ bytes so I went for it.

**Comment by EisenbergEffect** [comment]
I think we should probably hold on this for now until after the move to TS.

**Comment by EisenbergEffect** [comment]
Closing this for now. Let's revisit after the move to TS.

**Pull request closed without merging**